activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -910,14 +907,18 @@ def remove_connection(owner)
       # #fetch is significantly slower than #[]. So in the nil case, no caching will
       # take place, but that's ok since the nil case is not the common one that we wish
       # to optimise for.
```

Comment:
I noticed now that the `owner_to_pool.fetch` I’m seeing comes directly from the inlined `#pool_for` method. I’m unsure whether the `#fetch` here is as slow as the comment says.

Anyhow, either the comment or the code should be changed.